nixos/modules/services/web-apps/jitsi-meet.nix

```diff
@@ -0,0 +1,497 @@
+  } // (mapAttrs' (k: v: nameValuePair "-D${k}" v) cfg.videobridge.extraProperties);
+  jvbConfig = recursiveUpdate defaultJvbConfig cfg.videobridge.config;
+
+  jicofoArgs = {
+    "--host" = "localhost";
+    "--domain" = cfg.hostName;
+    "--secret" = "\${JICOFO_COMPONENT_SECRET}";
+    "--user_domain" = "auth.${cfg.hostName}";
+    "--user_name" = "focus";
+    "--user_password" = "\${JICOFO_USER_SECRET}";
+  };
+  jicofoProps = {
+    "-Dnet.java.sip.communicator.SC_HOME_DIR_LOCATION" = "/etc/jitsi";
+    "-Dnet.java.sip.communicator.SC_HOME_DIR_NAME" = "jicofo";
+    "-Djava.util.logging.config.file" = "/etc/jitsi/jicofo/logging.properties";
+    "-Djavax.net.ssl.trustStore" = "${dataDir}/prosody-ca.jks";
+    "-Djavax.net.ssl.trustStorePassword" = "javapls";
+  };
+  jicofoConfig = pkgs.writeText "sip-communicator.properties" (
+    concatStringsSep "\n" (mapAttrsToList (k: v: "${k}=${v}") cfg.jicofo.config)
+  );
+
+  # The configuration files are JS of format "var <<string>> = <<JSON>>;". In order to
+  # override only some settings, we need to extract the JSON, use jq to merge it with
+  # the config provided by user, and then reconstruct the file.
+  overrideJs =
+    source: varName: userCfg:
+    let
+      extractor = pkgs.writeText "extractor.js" ''
+        var fs = require("fs");
+        eval(fs.readFileSync(process.argv[2], 'utf8'));
+        process.stdout.write(JSON.stringify(eval(process.argv[3])));
+      '';
+      userJson = pkgs.writeText "user.json" (builtins.toJSON userCfg);
+    in (pkgs.runCommand "${varName}.js" { } ''
+      ${pkgs.nodejs}/bin/node ${extractor} ${source} ${varName} > default.json
+      (
+        echo "var ${varName} = "
+        ${pkgs.jq}/bin/jq -s '.[0] * .[1]' default.json ${userJson}
+        echo ";"
+      ) > $out
+    '');
+
+  # Essential config - it's probably not good to have these as option default because
+  # types.attrs doesn't do merging. Let's merge explicitly, can still be overriden if
+  # user desires.
+  defaultCfg = {
+    hosts = {
+      domain = cfg.hostName;
+      muc = "conference.${cfg.hostName}";
+    };
+    bosh = "//${cfg.hostName}/http-bind";
+  };
+
+  defaultJvbConfig = {
+    videobridge = {
+      apis.xmpp-client.configs.xmppserver1 = {
+        hostname = "localhost";
+        domain = "auth.${cfg.hostName}";
+        username = "jvb";
+        password = "__ENV__VIDEOBRIDGE_SECRET__ENV__";
+        muc_jids = "jvbbrewery@internal.${cfg.hostName}";
+        muc_nickname = builtins.replaceStrings [ "." ] [ "-" ] (
+          config.networking.hostName + optionalString (config.networking.domain != null) ".${config.networking.domain}"
+        );
+        disable_certificate_verification = true;
+      };
+      ice = {
+        tcp = {
+          enabled = true;
+          port = 4443;
+        };
+        udp.port = 10000;
+      };
+      stats = {
+        enabled = true;
+        transports = [ { type = "muc"; } ];
+      };
+    };
+  };
```

Comment:
This is a lot of magic that proably needs some explanatorily comment.

Reply:
Yes, it would probably be ideal to have as `default` for `services.jitsi-meet.videobridge.config` but not sure how because `types.attrs` don't do merging, so there's explicit merging using `recursiveUpdate`.

In future PR I'd like to expose the XMPP servers config (`apis.xmpp-client.configs.xmppserver1`) in the module configuration, that would hopefully make a bit it less magical. Added bunch of comments for now, let me know if anything is unclear.